src/drivers/mediatek/mt8186/interrupt.c

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Mediatek. All rights reserved.
+//
+// Author: Allen-KH Cheng <allen-kh.cheng@mediatek.com>
+//         Tinghan Shen <tinghan.shen@mediatek.com>
+
+#include <errno.h>
+#include <inttypes.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/spinlock.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define PENDING_IRQ_INDEX_MAX 32
+
+/* fa00558c-d653-4851-a03a-b21f125a9524 */
+DECLARE_SOF_UUID("irq-mt8186", irq_mt8186_uuid, 0xfa00558c, 0xd653, 0x4851,
```

Comment:
Need to generate another unique uuid for mt8186, don't use the same one with mt8195.